### PR TITLE
Fixes #627 : Refractors Codebase into folders

### LINKED ISSUE #627: Refactoring backend
Presently all the route handlers and app config are presently in a single file : backend/app/main.py. As a result, main.py is already quite big and as more route handlers and other methods will be added, the code will become really hard to maintain owing to the code size. So I guess it is high time now we modularise the APIs. 
We can use flask blueprint to distribute the API handlers among different modules. In future when we will be adding models (for database) we can add specific models to corresponding modules rather than keeping all the models in the same place.
Also we can maintain a separate file for config params and then initialise app.config from that file.
This will make the code easy to maintain, also we will be able to reuse stuff.

### PR DESCRIPTION
**Pull request by gabru-md**
Fixes Partially #627 

This PR contains the code for modularising the codebase.
This PR introduces the c oncepts of controllers and models. We can now go on into extending the 
current codebase into the new Flask Blueprint API.

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] I have added necessary documentation (if appropriate)

### Preview Link 

- **Replace XXX with your PR no**
- Link to live demo: http://pr-XXX-fossasia-badgeyay.surge.sh  

#### Changes proposed in this pull request:

- Introduces Modularised Codebase

Cheers!

### CODE REVIEW

**Comment by gabru-md** [comment]
@djmgit @ParthS007 @harshit98 please review

**Comment by djmgit** [comment]
@gabru-md so blueprint is a temporary folder right?

**Comment by djmgit** [comment]
@gabru-md I mean once the modules are ready they will be transfered under app right?

**Comment by gabru-md** [comment]
@djmgit Yes. We will work on it til it becomes stable. And then deploy it sideways as well. Once It serves the basic features then we will change the current backend eith these changes. :)

File: backend/blueprint/api/controllers/homePage.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+from flask import Blueprint, jsonify
+from utils.response import Response
+
+router = Blueprint('homePage', __name__)
+
+
+@router.route('/', methods=['POST'])
```
Comment (yashLadha) [inline backend/blueprint/api/controllers/homePage.py]: Why this is POST?

Comment (gabru-md) [reply backend/blueprint/api/controllers/homePage.py]: I was testing services while creating them using `cURL` therefore I kept it as `POST`

**Comment by djmgit** [comment]
@gabru-md so our db object (when we will be integrating db) will be kept in run.py right? This db object has to be imported from run.py into any model that will be present in api/models right? So can you once make sure whether you can import something from run.py into a file inside models?

File: backend/blueprint/api/controllers/homePage.py (hunk 1 above)
Comment (gabru-md) [reply backend/blueprint/api/controllers/homePage.py]: yes I and @ParthS007  will be working on this portion while you refractor them.

**Comment by gabru-md** [comment]
@djmgit Yes it is working perfectly fine.
We can import the database models from `models` folder into the `run.py` file.

**Comment by djmgit** [comment]
@gabru-md yes that is alright, but are we able to import anything from run.py to a file under api/models?

File: backend/blueprint/api/controllers/homePage.py (hunk 1 above)
Comment (yashLadha) [reply backend/blueprint/api/controllers/homePage.py]: Alright then. :+1:

**Pull request merged**